lib/deploy.rb

```diff
@@ -22,24 +24,42 @@ def run
       check_for_unstaged_changes!
       check_for_changelog!
       set_aws_region!
+      fetch_eb
       verify_configuration!
 
-      @name, @app_bucket = select_app
-      @platform          = detect_platform
+      @name       = deployment_target
+      @platform   = detect_platform
 
       request_confirmation!
       synchronize_repo!
       deploy!
     end
 
+    private
+
     def deploy!
       log 'Deployment commencing.'
       success = @platform.deploy!
-      abort "Deployment Failed, see system output." unless success
+      abort "Deployment Failed or timed out. See system output." unless success
       log 'All done.'
     end
 
-    private
+    def fetch_eb
+      @fetch_eb.nil? ? set_configuration_source : @fetch_eb
```

Comment:
I'm happy to update this BUT I can't use ||= with a boolean... I suppose the inelegant part is that though nil will have the same effect as false, I do want the value to be calculated if it nil, not if it is false.
Might have to rethink this part. Any thoughts?